contrib/terraform/gcp/kubernetes.tf

```diff
@@ -0,0 +1,182 @@
+  env_name          = "${var.env}"
+  gcp_project_id = "${var.gcp_project_id}"
+  component         = "ansible"
+  source            = "./modules/template"
+  template_name     = "${var.env}-ansible"
+  machine_type      = "${var.kube_ansible_machine_type}"
+  source_image      = "${var.kube_ansible_source_image}"
+  disk_size_gb      = "${var.kube_ansible_disk_size_gb}"
+  disk_type         = "${var.kube_ansible_disk_type}"
+  network_interface = "${var.kube_ansible_network_interface}"
+  subnetwork        = "${var.kube_ansible_subnetwork}"
+  mode              = "${var.kube_ansible_mode}"
+  svca_email        = "${var.kube_ansible_svca_email}"
+  svca_scopes       = "${var.kube_ansible_svca_scopes}"
+  target_size       = "${var.kube_ansible_target_size}"
+  region            = "${var.region}"
+}
+
+data "google_compute_instance" "ansible_ssh_machine_details" {
+  self_link = "${module.k8s_ansible_comp.ans_ssh_name}"
+}
+
+output "ansible_ssh_ip" {
+  value = "${data.google_compute_instance.ansible_ssh_machine_details.network_interface[0].network_ip}"
+}
+
+resource "null_resource" "ssh_ansible" {
+  connection {
+    host        = data.google_compute_instance.ansible_ssh_machine_details.network_interface[0].network_ip
+    type        = "ssh"
+    user        = var.user_name
+    timeout     = "500s"
+    private_key = file("${path.cwd}/kube_configurations/ssh-key")
+  }
+  provisioner "remote-exec" {
+    inline = [
+      "sudo yum install python-pip -y",
+      "sudo yum install git -y ",
+      "sudo pip install netaddr",
+      "sudo pip install logging",
+      "mkdir -p ${var.kube_automation_folder}",
+      "cd ${var.kube_automation_folder} && git clone ${var.kubespray_repo_url} && mv * kubespray",
+      "cd ${var.kube_automation_folder}/kubespray && sudo pip install -r requirements.txt"
+    ]
+  }
+  provisioner "file" {
+    source      = "${var.env}-inventory.ini"
+    destination = "${var.kube_automation_folder}/${var.env}-inventory.ini"
+  }
+
+  provisioner "file" {
+    source      = "${path.cwd}/kube_configurations/GenerateInventoryFile.py"
+    destination = "${var.kube_automation_folder}/GenerateInventoryFile.py"
+  }
+
+  provisioner "file" {
+    source      = "${path.cwd}/clustertfiles/gcp-auth-config-json/account.json"
+    destination = "${var.kube_automation_folder}/account.json"
+  }
+  provisioner "file" {
+    source      = "${path.cwd}/kube_configurations/ssh-key"
+    destination = "${var.kube_automation_folder}/key"
+  }
+  provisioner "remote-exec" {
+    inline = [
+      "cd ${var.kube_automation_folder} && chmod 400 key && chmod 777 GenerateInventoryFile.py && mv ${var.env}-inventory.ini inventory_hosts.ini",
+      "cd ${var.kube_automation_folder} && python GenerateInventoryFile.py ${var.kube_automation_folder} ${var.user_name}",
+      "cd ${var.kube_automation_folder}/kubespray && ansible-playbook -i inventory/cluster/inventory_hosts.ini -b -v cluster.yml 2>&1 | tee outfile",
```

Comment:
Ansible playbook can take the dynamic inventory directly without using a temporary file. Something like `ansible-playbook -i myscript.py`